### PR TITLE
App variables reduced and removed

### PR DESCRIPTION
**Pull request by annando**
This PR is slightly huge. The `App` class is the home of a lot of variables that are shared across the system. The process often isn't very transparent or debugable.

Now many of these variables are removed or at least reduced.

This PR currently is a draft since I want to test it for a while - and possibly will add some more stuff to it.

### CODE REVIEW

**Comment by annando** [comment]
@MrPetovan what about using session variables for some of the `App` variables?

**Comment by annando** [comment]
@MrPetovan @nupplaphil I had no issues the last few hours. So I think that the code is okay.

File: mod/display.php [hunk 1]
```diff
@@ -109,41 +110,28 @@ function display_init(App $a)
...
 	if (!empty($shared) && empty($shared['comment'])) {
+		$profiledata = [];
+		$profiledata['uid'] = -1;
+		$profiledata['id'] = -1;
+		$profiledata['nickname'] = '';
+		$profiledata['name'] = '';
+		$profiledata['picdate'] = '';
+		$profiledata['photo'] = '';
+		$profiledata['url'] = '';
+		$profiledata['network'] = '';
```
Comment (MrPetovan) [inline mod/display.php]:
```suggestion
		$profiledata = [
			'uid' => -1,
			'id' => -1,
			'nickname' => '',
			'name' => '',
			'picdate' => '',
			'photo' => '',
			'url' => '',
			'network' => '',
		];
```

File: mod/photos.php [hunk 2]
```diff
@@ -1636,7 +1624,7 @@ function photos_content(App $a)
 	$o .= Renderer::replaceMacros($tpl, [
 		'$title' => DI::l10n()->t('Recent Photos'),
 		'$can_post' => $can_post,
-		'$upload' => [DI::l10n()->t('Upload New Photos'), 'photos/'.$a->data['user']['nickname'].'/upload'],
+		'$upload' => [DI::l10n()->t('Upload New Photos'), 'photos/'.$user['nickname'].'/upload'],
```
Comment (MrPetovan) [inline mod/photos.php]:
```suggestion
		'$upload' => [DI::l10n()->t('Upload New Photos'), 'photos/' . $user['nickname'] . '/upload'],
```

File: src/App.php [hunk 3]
```diff
@@ -547,7 +617,7 @@ public function runFrontend(App\Module $module, App\Router $router, IPConfig $pc
 				$this->baseURL->redirect('search');
 			}
 
-			// Initialize module that can set the current theme in the init() method, either directly or via App->profile_uid
+			// Initialize module that can set the current theme in the init() method, either directly or via App->profile_owner
```
Comment (MrPetovan) [inline src/App.php]: Shouldn't we mention the setter method instead?

File: src/Module/Contact.php [hunk 4]
```diff
@@ -453,9 +445,7 @@ public static function content(array $parameters = [], $update = 0)
 
 		$_SESSION['return_path'] = DI::args()->getQueryString();
 
-		if (!empty($a->data['contact']) && is_array($a->data['contact'])) {
-			$contact = $a->data['contact'];
-
+		if (!empty($contact) && is_array($contact)) {
```
Comment (MrPetovan) [inline src/Module/Contact.php]: Why `is_array()`? What else can `$contact` be at this point?

File: src/Module/Delegation.php [hunk 5]
```diff
@@ -118,7 +118,7 @@ public static function content(array $parameters = [])
 			throw new ForbiddenException(DI::l10n()->t('Permission denied.'));
 		}
 
-		$identities = DI::app()->identities;
+		$identities = User::identities(DI::session()->get('submanage') ?: local_user());
```
Comment (MrPetovan) [inline src/Module/Delegation.php]: `Session->get()` already has a fallback: 
```suggestion
		$identities = User::identities(DI::session()->get('submanage', local_user()));
```

File: src/Module/Magic.php [hunk 6]
```diff
@@ -65,7 +65,7 @@ public static function init(array $parameters = [])
 		$contact = DBA::selectFirst('contact', ['id', 'nurl', 'url'], ['id' => $cid]);
 
 		// Redirect if the contact is already authenticated on this site.
-		if (!empty($a->contact) && array_key_exists('id', $a->contact) && strpos($contact['nurl'], Strings::normaliseLink(DI::baseUrl()->get())) !== false) {
+		if (!empty($a->getContactId()) && strpos($contact['nurl'], Strings::normaliseLink(DI::baseUrl()->get())) !== false) {
```
Comment (MrPetovan) [inline src/Module/Magic.php]: Redundant check:
```suggestion
		if ($a->getContactId() && strpos($contact['nurl'], Strings::normaliseLink(DI::baseUrl()->get())) !== false) {
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Contact.php (hunk 4 above)
Comment (annando) [reply src/Module/Contact.php]: In many cases I did a blind search and replace without judging the code.

File: src/App.php (hunk 3 above)
Comment (annando) [reply src/App.php]: Possibly.

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
> @MrPetovan what about using session variables for some of the `App` variables?

It's worse since it's even harder to look for session variable access/setting. Session variables should only be used for data that needs to be persisted between pages, App variables were exclusively meant for data that needs to be persisted in a single page call.

Ultimately, I believe each App variable should be attached to the relevant class but what you did with setters/getters in App is fine for now.

**Comment by annando** [comment]
@MrPetovan in fact I would like to get rid of the `App` class in total.

**Comment by MrPetovan** [comment]
Address https://github.com/friendica/friendica/issues/10473#issuecomment-886077702

**Pull request merged**